server/neptune/gateway/event/pull_request_review_handler.go

```diff
@@ -29,25 +35,43 @@ type fetcher interface {
 	ListFailedPolicyCheckRunNames(ctx context.Context, installationToken int64, repo models.Repo, ref string) ([]string, error)
 }
 
-type PullRequestReviewWorkerProxy struct {
-	Scheduler       scheduler
-	SnsWriter       Writer
-	Logger          logging.Logger
-	CheckRunFetcher fetcher
+type prApprovalSignaler interface {
+	SignalWorkflow(ctx context.Context, workflowID string, runID string, signalName string, arg interface{}) error
 }
 
-func (p *PullRequestReviewWorkerProxy) Handle(ctx context.Context, event PullRequestReview, request *http.BufferedRequest) error {
-	return p.Scheduler.Schedule(ctx, func(ctx context.Context) error {
-		return p.handle(ctx, event, request)
-	})
+type PullRequestReviewWorkerProxy struct {
+	Scheduler          scheduler
+	SnsWriter          Writer
+	Logger             logging.Logger
+	CheckRunFetcher    fetcher
+	Allocator          feature.Allocator
+	PRApprovalSignaler prApprovalSignaler
+	Scope              tally.Scope
 }
 
-func (p *PullRequestReviewWorkerProxy) handle(ctx context.Context, event PullRequestReview, request *http.BufferedRequest) error {
+func (p *PullRequestReviewWorkerProxy) Handle(ctx context.Context, event PullRequestReview, request *http.BufferedRequest) error {
 	// Ignore non-approval events
 	if event.State != Approved {
 		return nil
 	}
+	var combinedErrors error
+	err := p.Scheduler.Schedule(ctx, func(ctx context.Context) error {
+		return p.handleLegacyMode(ctx, request, event)
+	})
+	if err != nil {
```

Comment:
according to the docs:

```
Any nil errors within errs will be ignored.
```

So you can just append the resulting err without this check.

Reply:
Testing showed me that they still treat the end result as an error, but an error with no underlying errors stored inside